docs/en/examples/elementalarea_with_dataobject.md

```diff
@@ -0,0 +1,161 @@
+
+```php
+<?php
+
+namespace App\Admins;
+
+use App\Models\BlogPost;
+use SilverStripe\Admin\ModelAdmin;
+
+class BlogPostsAdmin extends ModelAdmin
+{
+
+    private static string $url_segment = 'blog-posts-admin';
+
+    private static string $menu_title = 'Blog Posts';
+
+    private static string $menu_icon_class = 'font-icon-block-banner';
+
+    private static array $managed_models = [
+        BlogPost::class,
+    ];
+}
+```
+
+`$managed_models` - this static variable connects BlogPostAdmin with BlogPost, which will be created in the following code.
+When you create a DataObject you should define the connection between your DataObject and ElementalArea by adding `$has_one` relationship.
+In this example BlogPost will have one ElementalArea.
+
+**app/src/Models/BlogPost.php**
+
+```php
+<?php
+
+namespace App\Models;
+
+use App\Admins\BlogPostsAdmin;
+use DNADesign\Elemental\Models\ElementalArea;
+use SilverStripe\Control\Controller;
+use SilverStripe\ORM\DataObject;
+use SilverStripe\Versioned\Versioned;
+
+class BlogPost extends DataObject
+{
+    private static string $table_name = 'BlogPost';
+
+    private static $extensions = [
+        Versioned::class,
+    ];
+
+    private static array $db = [
+        'Title' => 'Varchar(255)',
+        'Content' => 'HTMLText',
+    ];
+
+    private static $has_one = [
+        'ElementalArea' => ElementalArea::class,
+    ];
+
+    private static $owns = ['ElementalArea'];
+
+    // ...
+}
+ ```
+
+If you are using ElementalArea together with DataObject, it is important to define a few following methods in the class.
+First of all, the CMSEditLink() method should be defined to generate a link for opening the editing section of the particular BlogPost.
+BaseElement uses a link, that was created in CMSEditLink() method in BlogPost class, to generate new links for opening the editing section of the particular Element.
+
+```php
+// ...
+
+class BlogPost extends DataObject
+{
+    // ...
+
+    public function CMSEditLink()
+    {
+        // In this example we use BlogPostsAdmin class as Controller
+        $admin = BlogPostsAdmin::singleton();
+
+        // Makes link more readable. Instead App\Models\ BlogPost, we get App-Models-BlogPost
+        $sanitisedClassname = str_replace('\\', '-', $this->ClassName);
+
+        // Returns link to editing section with elements
+        return Controller::join_links(
+            $admin->Link($sanitisedClassname),
+            'EditForm/field/',
+            $sanitisedClassname,
+            'item',
+            $this->ID,
+        );
+    }
+
+    // ...
+}
+
+```
+
+Method Link() doesn't exist in DataObject class, so you should define this method in your class. You should add the canArchive() method in your class as well to set rules for the canDelete() method.
+You should also define the AbsoluteLink() method for opening the tabs in the Elements.
+Use the following code as an example of how to simply create these methods.
```

Comment:
```suggestion
The `Link()` method doesn't exist in the`DataObject` class, so you should define this method in your class. You should add the `canArchive()` method in your class as well to set rules for the `canDelete()` method.
You should also define the `AbsoluteLink()` method for opening the tabs in the elements.
Use the following code as an example of a simple implementation of these methods.
```

Reply:
Why is `Link()` needed? If this is just for preview purposes, we can leave it out of this documentation and just refer to the new preview documentation.
Also what do you mean by "opening the tabs in the elements"?

Reply:
Since BaseElement implemented CMSPreviewable before (in this version CMSPreviewable interface was removed from  the BaseElement class), we have :
```php
                HiddenField::create('LiveLink', false, Director::absoluteURL($this->Link())),
                HiddenField::create('StageLink', false, Director::absoluteURL($this->PreviewLink())),
  ```
  in public function getCMSFields(). I mentioned this before and I also said that we should refactor BaseElement class and remove all unused methods.

Reply:
These methods are used in the next minor so that's fine.
If the issue is that `BaseElement::Link()` calls `$this->getPage()->Link()` then there should be a check in there. Something like this:

```php
if (ClassInfo::hasMethod($page, 'Link')) {
    $link = $page->Link($action);
    // do any transformations on the link e.g. concatenations
    return $link;
}

return null;
```

That way we're not imposing a link on the dataobject where it doesn't make sense to have one.

Reply:
If we have Element that has additional tabs in Element, tab itself has link like "http://localhost.com/admin/blog-posts-admin/App-Models-BlogPost/EditForm/field/App-Models-BlogPost/item/4/ItemEditForm/field/ElementalArea/item/9/edit**#Root_Settings**"

Reply:
Oh, you mean tabs in the CMS edit form! Okay. That won't come from the `Link()` method on the `Page`  though, surely. Otherwise it would looks like this for blocks pages:
`http://localhost.com/some-page-path/ItemEditForm/field/ElementalArea/item/9/edit#Root_Settings`
I think this is coming from the form itself, rather than relying on some `Link()` method to be defined on the `DataObject` subclass that owns the element. I could even see it _maybe_ using `CMSEditLink`, but `Link()` is for front-end purposes.

Reply:
I've tested this locally without `Link()` on the `BlogPost` class - so long as there's a check like the one I mention above it works fine even with tabs in the CMS.